src/resolve_type.hack

```diff
@@ -12,12 +12,35 @@ namespace Facebook\HHAST;
 use namespace HH\Lib\C;
 use namespace Facebook\HHAST\__Private\Resolution;
 
-function resolve_type(string $type, Script $root, Node $node): string {
+enum ResolvedTypeKind: string {
+  GENERIC_PARAMETER = 'generic_parameter';
+  QUALIFIED_TYPE = 'qualified_type';
+}
+
```

Comment:
Please split this to a separate file

Reply:
Hmm I'm OK either way but in general I don't really like putting everything in separate files -- it feels like taking a class and putting each method in a different file :/